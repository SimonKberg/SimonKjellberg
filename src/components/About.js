// @flow strict
import * as React from 'react'
import getConfig from 'next/config'
import Heading from './Heading'

const {
  publicRuntimeConfig: { siteDescription },
} = getConfig()

const About = () => (
  <section>
    <Heading level="2">About</Heading>
    <p>{siteDescription}</p>
  </section>
)

export default About
