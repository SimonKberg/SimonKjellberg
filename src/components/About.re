[@bs.config {jsx: 3}];

let str = React.string;
let config = Next.Config.getConfig();

[@react.component]
let make = () => {
  <section>
    <Heading level=`Level2>
      {"About " |> str}
      <Subtitle> {"(Location: Stockholm, Sweden)" |> str} </Subtitle>
    </Heading>
    <p> {config##publicRuntimeConfig##siteDescription} </p>
    <p>
      {"Working as a consultant via my own company, " |> str}
      <Link href="https://shebang.consulting"> {"Shebang" |> str} </Link>
      {"." |> str}
    </p>
  </section>;
};

let default = make;
