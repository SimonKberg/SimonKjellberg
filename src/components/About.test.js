// @flow strict
import * as React from 'react'
import renderer from 'react-test-renderer'
import About from './About'

jest.mock('./Heading', () => 'Heading')

test('<About />', () => {
  const tree = renderer.create(<About />).toJSON()
  expect(tree).toMatchSnapshot()
})
