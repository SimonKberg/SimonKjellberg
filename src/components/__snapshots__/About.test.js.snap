// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<About /> 1`] = `
<section>
  <Heading
    level="2"
  >
    About
  </Heading>
  <p>
    Fullstack web developer, specialized in React, Node.js, and GraphQL, with a strong focus on building scalable frontend architecture.
  </p>
</section>
`;
